### ISSUE TITLE
Lack of overall maintenance efforts or abandoned?

### ISSUE THREAD

**Issue by mmzx** [snoyberg/keter#201]
Hi All, 

For a while I did not need to install **keter**, but now I would need it and it seems to have several wounds. Is the project abandoned or would simply just benefit from more regular maintenance? 

I would be very happy to help if there is demand for it!

Like,
- *master* brach is failiing to build currently
- CI script has old ghc versions in it
- maybe there is something useful in the existing pull-requests and some of them could be merged in?

**Comment by tolysz** [comment]
I still use it. The challenge is once you build it, there is no need to touch it.
For sure there is something useful in the PRs :P

**Comment by mmzx** [comment]
I am setting up a new server at the moment and ended up copying the _keter_ binary from the old server. - which is not ideal. On the current version of FreeBSD the ghc version is actually unavailable (ghc-8.0.2 was used last time); so basically there is no chance I could compile _keter_ on a recent version of this OS, well including Linux as well.

Again, I am happy to work on new PRs as long as the effort makes sense.

**Comment by tolysz** [comment]
You could try to build from the:
https://github.com/snoyberg/keter/pull/195 ( https://github.com/tolysz/keter/tree/modernize )

**Comment by mmzx** [comment]
Compiles. Great. Thanks. :) 

As a contributor you would be able to merge onto master, right?

**Comment by mmzx** [comment]
I have started making small modifications: 
@tolysz , baseed on your branch **modernize** I have found the followings:

- keter's code is written using Internals of the _process_ library. Which changes actually quiet a bit between versions, so I think it is not a  good idea to use the _Internal_ parts of the library. There are unhandled pattern-match warnings because of these changes of the internal API. By pinning process-1.4.3.0 to which originally was written against these issues are resolved easily without any extra consequence.

- additonally some easy fix is also provided.

- On top of these by having a new working CI script multiple ghc  versions could be tested I think.

- By pinning process-1.4.3.0 for the time being newer lts versions could be used if needed.

**Comment by tolysz** [comment]
Contributor is a label for people who ever got merged into the code... nothing more. You need people with "Collaborator" labels :)

**Comment by mmzx** [comment]
Right! I keep putting effort into it, hopefully someone will listen up...

**Comment by tolysz** [comment]
The `modernize` branch builds with `LTS Haskell 15.0`... no need to pin process yet...

**Comment by mmzx** [comment]
The `modernize` branch lacks stack.yaml, so that definitely brakes the builds using stack.

**Comment by tolysz** [comment]
`stack init && stack setup && stack install`

**Comment by mmzx** [comment]
> `stack init && stack setup && stack install`
> possibly before all that
> `stack upgrade --git`

Hm. Doing this would be against having pre-defined lts versions: and/or will require lts specific stack configuration files which results in no need to run `stack init && stack setup`.

**Comment by mmzx** [comment]
In #203 it is realized to have resolver specific stack files. Initializing stack config isn't needed any more, indeed it actually may fall back to a too old LTS version in some cases.
I will start adding resolvers and apply fixes to warnings.

**Comment by vaclavsvejcar** [comment]
@snoyberg It looks there's still some need for this tool, I also considered using that but on the other hand I don't want to use abandonned software. Do you still plan to maintain this tool? Thanks.

**Comment by snoyberg** [comment]
I passed maintenance of Keter over to @creichert a few years back. I don't know if he is still actively maintaining. If not, and if others want to step up as maintainers, I'm happy to have out maintainer bits.

**Comment by mmzx** [comment]
Please, please please! Volunteering.

**Comment by lucch** [comment]
There is definitely some need for this tool! 🙂

**Comment by mmzx** [comment]
Cross reference to #203.

**Comment by creichert** [comment]
Hey all, I am not actively using Keter anymore so of course my maintenance efforts have diminished as my time is required elsewhere.

I will do my best to help any new maintainers come on board and I'm happy to review PRs.

**Comment by snoyberg** [comment]
@mmzx @lucch I've invited both of you as collaborators on this repository.

**Comment by benjamingoldstein** [comment]
It is good to see this happening. 

Did you all get the project back up and running again? I can't seem to run the install instruction command, as per issue #206

**Comment by mmzx** [comment]
@jappeace , thanks for the direct call-out! I am going to look at the PRs...

**Comment by jezen** [comment]
@snoyberg We're using keter regularly at Supercede. Any chance we could also be maintainers of this?

**Comment by snoyberg** [comment]
Absolutely, invite sent. Let me know if you want other people on your team to have access too.

**Comment by jezen** [comment]
Thanks Michael. The more, the merrier! Relevant people include:

- @jappeace
- @ivb-supercede
- @timds 
- @cevantes 
- @NikitaRazmakhnin 
- @VarmaMSP 
- @LukeTemp.

**Comment by snoyberg** [comment]
All invited

**Comment by jappeace** [comment]
Could you also make someone hackage maintainer so we can make a release? I have a bunch of nix code I wish to integrate in nixpkgs, that'd be a lot easier if there was a new hackage release. It's possible to run Keter from nixos with that code.

**Comment by snoyberg** [comment]
Sure, @jezen can you send me your Hackage username?

**Comment by jezen** [comment]
Yep! My username on Hackage is [`jgt`](https://hackage.haskell.org/user/jgt).

Cheers!

**Comment by snoyberg** [comment]
Added :)

**Comment by jezen** [comment]
There's nothing else to do here, so I'll close this issue.

Future readers: Want to help with maintaining keter? [Open an issue](https://github.com/snoyberg/keter/issues/new) 🙂
